core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1209,6 +1227,29 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
     }
   }
 
+  case class BrokerModifications() extends ControllerEvent {
+    override def state: ControllerState = ControllerState.BrokerChange
+
+    override def process(): Unit = {
+      if (!isActive) return
+      val curBrokers = zkClient.getAllBrokersInCluster.toSet
+      val updatedBrokers = controllerContext.liveBrokers.filter { broker =>
+        val existingBroker = curBrokers.find(_.id == broker.id)
+        existingBroker match {
+          case Some(b) => broker.endPoints != b.endPoints
+          case None => false
+        }
+      }
+      val updatedBrokerIdsSorted = updatedBrokers.map(_.id).toSeq.sorted
```

Comment:
Maybe we can move this into the `if` below and use check `updatedBrokers.nonEmpty` instead? It avoids the log message when there are no updated brokers.